crates/swc_ecma_lints/src/config.rs

```diff
@@ -0,0 +1,41 @@
+use crate::rules::no_console::NoConsoleConfig;
+use serde::{Deserialize, Serialize};
+use std::fmt::Debug;
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "lowercase")]
+pub enum LintRuleReaction {
+    Off,
+    Warning,
+    Error,
+}
+
+impl Default for LintRuleReaction {
+    fn default() -> Self {
+        Self::Off
+    }
+}
+
+#[derive(Debug, Clone, Default, Serialize, Deserialize)]
+pub struct RuleConfig<T: Debug + Clone + Serialize + Default>(
+    #[serde(default)] LintRuleReaction,
+    #[serde(default)] T,
+);
+
+impl<T: Debug + Clone + Serialize + Default> RuleConfig<T> {
+    pub(crate) fn get_rule_reaction(&self) -> &LintRuleReaction {
+        &self.0
+    }
+
+    pub(crate) fn get_rule_config(&self) -> &T {
+        &self.1
+    }
+}
+
+#[derive(Debug, Clone, Default, Serialize, Deserialize)]
+#[non_exhaustive]
+#[serde(rename_all = "camelCase")]
```

Comment:
For better compatibility with the existing lint config, it is better to use the `kebab-case`.
@ArturAralin @kdy1

Reply:
Maybe we can add lots of aliases instead

Reply:
Aliases bring more boilerplate code